src/create-matcher.js

```diff
@@ -198,6 +203,26 @@ function fillParams (
   }
 }
 
+function getParams (path: string): Array<string> {
+  const hit = regexpParamsCache[path]
+
+  if (hit) {
+    return hit
+  }
+
+  const keys = []
+  const paramNames: Array<string> = []
+
+  Regexp(path, keys)
+
+  for (let i = 0, len = keys.length; i < len; ++i) {
+    const key = keys[i]
+    if (key && 'name' in key) paramNames.push(key.name)
+  }
+
+  return (regexpParamsCache[path] = paramNames)
```

Comment:
This can be simplified to 
```js
return regexpParamsCache[path] || 
   (regexpParamsCache[path] = keys.map(k => k.name))
```
like https://github.com/znck/vue-router/blob/ab0a779ba6f09f5b28138c9445ce84d7263daef6/src/create-matcher.js#L196-L198